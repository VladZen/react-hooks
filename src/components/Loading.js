import '@styles/components/loading.scss'

const Loading = ({ trigger, children, ...props }) => {
  const properties = {
    ...props,
    className: props.className
      ? `spinner ${props.className}`
      : 'spinner'
  }

  return trigger
    ? <div {...properties}>Loading...</div>
    : children
}

export default Loading
