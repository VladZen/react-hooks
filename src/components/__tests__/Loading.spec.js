import { unmountComponentAtNode, render } from 'react-dom'
import { act } from 'react-dom/test-utils'
import Loading from '@/components/Loading'

let container = null
beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  unmountComponentAtNode(container)
  container.remove()
  container = null
})

describe('Loading component', () => {
  describe('rendering', () => {
    it('shows spinner', () => {
      act(() => {
        render(
          <Loading trigger={true}>
            <a href='#'>
              Some link
            </a>
          </Loading>,
          container
        )
      })

      expect(container).toMatchSnapshot()
    })

    it('shows content', () => {
      act(() => {
        render(
          <Loading trigger={false}>
            <a href='#'>
              Some link
            </a>
          </Loading>,
          container
        )
      })

      expect(container).toMatchSnapshot()
    })

    it('shows spinner with applied extra props', () => {
      act(() => {
        render(
          <Loading
            className='additional-class'
            trigger={true}
          >
            <a href='#'>
              Some link
            </a>
          </Loading>,
          container
        )
      })

      expect(container).toMatchSnapshot()
    })
  })
})
