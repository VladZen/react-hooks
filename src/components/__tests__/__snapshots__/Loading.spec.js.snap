// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component rendering shows content 1`] = `
<div>
  <a
    href="#"
  >
    Some link
  </a>
</div>
`;

exports[`Loading component rendering shows spinner 1`] = `
<div>
  <div
    class="spinner"
  >
    Loading...
  </div>
</div>
`;

exports[`Loading component rendering shows spinner with applied extra props 1`] = `
<div>
  <div
    class="spinner additional-class"
  >
    Loading...
  </div>
</div>
`;
